alacritty/src/display/hint.rs

```diff
@@ -70,21 +93,83 @@ impl HintState {
             None => return,
         };
 
-        // Find visible matches.
-        self.matches.0 = hint.regex.with_compiled(|regex| {
-            let mut matches = RegexMatches::new(term, regex);
-
-            // Apply post-processing and search for sub-matches if necessary.
-            if hint.post_processing {
-                matches
-                    .drain(..)
-                    .map(|rm| HintPostProcessor::new(term, regex, rm).collect::<Vec<_>>())
-                    .flatten()
-                    .collect()
-            } else {
-                matches.0
+        let mut hint_matches: Vec<HintMatchLocation> = Vec::new();
```

Comment:
This method is an absolute massive mess and I'm going to hold off on a review until it's cleaned up.

Reply:
The traversing part is separated to `traverse_hyperlinks` now.